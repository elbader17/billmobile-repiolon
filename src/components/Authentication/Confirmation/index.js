import { connect } from 'react-redux';
import Confirmation from './Component';
import { confirmCode } from '../../../app/authentication/actions';

const mapStateToProps = () => ({});

function mapDispatchToProps(dispatch) {
  return {
    confirmCode: (email, confirmationCode) => dispatch(confirmCode(email, confirmationCode)),
  };
}

const component = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Confirmation);

export default component;
