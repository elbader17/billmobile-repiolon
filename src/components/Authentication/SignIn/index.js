import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Input, Button } from "react-native-elements";
import { signIn } from '../../../app/authentication/actions';

class SignIn extends React.Component {

    constructor(props) {
      super(props);
      this.state = {
        email: '',
        password: '@Am1234!',
        name:'',
        confirmPassword: '@Am1234!',
      };
    }

    handleSignIn = () => {
      const { email, password } = this.state;
      const { dispatch } = this.props;
      dispatch(signIn(email, password));
    }

    setEmail = (value) => this.setState({ email: value })
    setPassword = (value) => this.setState({ password: value })

    render() {
      return(
          <View>
            <Input
              label="Email"
              leftIcon={{ type: 'font-awesome', name: 'envelope' }}
              onChangeText={this.setEmail}
              placeholder="my@email.com"
            />
            <Input
              label="Password"
              leftIcon={{ type: 'font-awesome', name: 'lock' }}
              onChangeText={this.setPassword}
              placeholder="p@ssw0rd123"
              secureTextEntry
            />
            <Button
              title='Submit'
              onPress={ this.handleSignIn }
            />
        </View>
      )
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});


export default SignIn;
