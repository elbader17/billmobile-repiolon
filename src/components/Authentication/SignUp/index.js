import { connect } from 'react-redux'
import SignUp from './Component';
import { signUp } from '../../../app/authentication/actions';
import { confirmCode } from '../../../app/authentication/actions';

const mapStateToProps = state => ({
  showConfirmationModal: state.authentication.showConfirmationModal,
});

function mapDispatchToProps(dispatch) {
  return {
    signUp: (password, name,attributes) => dispatch(signUp(password, name, attributes)),
  };
}

const component = connect(
  mapStateToProps,
  mapDispatchToProps,
)(SignUp);

export default component;
