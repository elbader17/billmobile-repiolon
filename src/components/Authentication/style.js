import { StyleSheet } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { COLORS } from '../../constants/colors';
import { FONTS } from '../../constants/fonts';

const styles = StyleSheet.create({
  container: {
    height: hp('100%'),
  },
  containerBody: {
    flex: 0.90
  },
  containerFooter: {
    flex: 0.10,
    justifyContent: 'flex-end'
  },
  containerHeader: {
    paddingVertical: 25,
    alignItems: 'center',
    backgroundColor: COLORS.white,
  },
  containerInputs: {
    flex: 1,
    marginHorizontal: 28
  },
  containerButtonSignTwo: {
    alignItems: 'center', 
    marginTop: 18,
  },
  gradientStyle: { 
    width: wp('50%'),
    height: hp('5%'),
    borderRadius: 20,
    marginTop: 7,
  },
  textSubLogo: {
    flex:1,
    marginTop: 5,
    marginBottom: 5,
    marginLeft: 35,
    justifyContent: 'center',
    alignItems: 'center'
  },
  textCenter: {
    flex:1,
    marginTop: 10,
    justifyContent: 'center',
    alignItems: 'center'
  },
  textRegular14White: {
    fontFamily: FONTS.pRegular,
    fontSize: FONTS.size14,
    color: COLORS.white,
    lineHeight: 22,
    textAlign: 'center'
  },
  textRegular16White: {
    fontFamily: FONTS.pRegular,
    fontSize: FONTS.size16,
    color: COLORS.white,
    textAlign: 'center',
    justifyContent: 'center',
    top: 1
  },
  imageHeader: {
    width: 219,
    height: 130
  },
  buttonSign: {
    width: wp('40%'),
    height: hp('6%'),
    backgroundColor: COLORS.blue,
    borderRadius: 25,
    justifyContent: 'center',
  },
  buttonSignDisable: {
    width: wp('42%'),
    height: hp('6%'),
    backgroundColor: COLORS.gray,
    borderRadius: 25,
    justifyContent: 'center',
  },
  buttonSignTwo: {
    width: wp('86%'),
    height: hp('7%'),
    backgroundColor: COLORS.blue,
    borderRadius: 25
  },
  buttonVerify: {
    width: wp('86%'),
    height: hp('7%'),
    backgroundColor: 'transparent',
    borderRadius: 25,
    marginTop: 35,
    elevation: 2
  },
  buttonConfirm: {
    width: wp('50%'),
    height: hp('6%'),
    backgroundColor: 'transparent',
    borderRadius: 25,
    marginTop: 15,
    elevation: 1
  },
  gradientBottom: { 
    backgroundColor: COLORS.blue,
    height: hp('9%'),
    borderTopLeftRadius: 100, 
    borderTopRightRadius: 100, 
    justifyContent: 'center',
  },
  textMedium16Blue: {
    fontFamily: FONTS.pRegular,
    fontSize: FONTS.size14,
    color: COLORS.blue
  },
  textRegular18BlueMedium: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size18,
    color: COLORS.blueMedium
  },
  textRegular11White: {
    fontFamily: FONTS.pRegular,
    fontSize: FONTS.size11,
    color: COLORS.white,
    textAlign: 'center',
    top: 2
  },
  textRegular11WhiteBold: {
    fontFamily: FONTS.pSemiBold,
    fontSize: FONTS.size11,
    color: COLORS.blueLight,
    textAlign: 'center',
  },
  textRegular12GrayDark: {
    fontFamily: FONTS.pRegular,
    fontSize: FONTS.size11,
    color: COLORS.grayDark,
    textAlign: 'center'
  },
  inLineSpaceAround: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center'
  },
  inLine: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  displayNone: {
    display: 'none'
  },
  displayFlex: {
    display: 'flex'
  }
});

export default styles;