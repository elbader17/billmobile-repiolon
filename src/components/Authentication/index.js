import React from 'react';
import { View, Text, Image, ScrollView} from 'react-native';
import { Button } from "react-native-elements";
import Signup from './SignUp';
import Signin from './SignIn';
import style from './style';

class Authentication extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      selectedSignUp: false, 
    };
  }

  renderSignUpSignIn = () => {  
    if (this.state.selectedSignUp) {
      return (
        <View>
          <Signup/>
        </View>
      );
    }else {
      return (
        <View>
          <Signin/>
        </View>
      );
    }
  }

  render() {
    const logo = require('../../images/iconBill.png')
    const buttonOn = [style.button,style.textRegular12BlueBold];
    const buttonOff = [style.buttonDisabled,style.textRegular12WhiteBold];
    return(
      <ScrollView>
          <View style={style.container}>
            <View style={ style.containerHeader }>
              <Image source={ logo } style={ style.imageHeader } />
              <Text style={[style.textRegular14White, {paddingTop: 10}]}>{"Hacé facturas electronicas rápido"}.{"\n"} 
              Y hacelo seguro</Text>
            </View>
            <View style={style.inLine}>
              <Button
                title='REGISTRAR'
                testID='register'
                onPress={() => this.setState({selectedSignUp: true})}
                buttonStyle={ this.state.selectedSignUp ? buttonOn[0] : buttonOff[0] }
                titleStyle={ this.state.selectedSignUp ? buttonOn[1] : buttonOff[1] }
              />
              <Button
                title='INICIAR SESIÓN'
                testID='signin'
                onPress={() => this.setState({selectedSignUp: false})}
                buttonStyle={ this.state.selectedSignUp ? buttonOff[0] : buttonOn[0] }
                titleStyle={ this.state.selectedSignUp ? buttonOff[1] : buttonOn[1] }
              />
            </View>
            {this.renderSignUpSignIn()} 
            <View style={style.containerFooter}>
              <Text style={style.textRegular11GrayDark}>
                Al registrarte estas aceptando nuestros
              </Text>
              <Text style={style.textRegular11GrayDarkBold}>
                Términos y Condiciones y Políticas de Privacidad
              </Text> 
            </View>
          </View>
      </ScrollView>
    );
  }
}

export default Authentication;