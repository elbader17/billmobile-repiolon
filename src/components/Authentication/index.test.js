import React from 'react';
import { shallow } from 'enzyme';
import Authentication from './index';

it('matches the snapshot', () => {
  const tree = shallow(<Authentication />);
  expect(tree).toMatchSnapshot();
});


