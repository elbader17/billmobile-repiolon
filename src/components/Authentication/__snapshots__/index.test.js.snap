// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Component Authentication renders correctly Authentication 1`] = `ShallowWrapper {}`;
